remotes/docker/credentials/docker_credential_helper.go

```diff
@@ -0,0 +1,101 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package credentials
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"strings"
+)
+
+// dockerCredentialHelper provides implementation for a credential helper that's compatible to the
+// docker-credential-helpers (https://github.com/docker/docker-credential-helpers) executables.
+type dockerCredentialHelper struct {
+	path string
+}
+
+func (p *dockerCredentialHelper) Get(serverURL string) (*Credentials, error) {
+	in := strings.NewReader(serverURL)
+	response, err := p.exec(in, "get")
```

Comment:
Can use https://github.com/docker/docker-credential-helpers/blob/master/client/client.go directly?

Reply:
The reason for not using that client is because we want to add the support for setting static authorization header so the `Credentials` struct is defined as

```go
type Credentials struct {
	// ServerURL is the host name of the server
	ServerURL string
	// Username is the username used to authenticate to the server
	Username string
	// Secret is the secret for the username to authenticate to the server
	Secret string
	// Header is the authorization value to be set in the header for the http request
	Header string
}
```

Whereas the client from `docker-credential-helper` does not include this `Header` field
https://github.com/docker/docker-credential-helpers/blob/master/credentials/credentials.go#L14

With this, we can still be compatible with `docker-credential-helper` but can also support custom plugins that implement the `docker-credential-helper` cli but could add the extra static authorization header field.